core-plugins/src/test/java/io/cdap/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -374,6 +373,62 @@ public void testPathGlobbing() throws Exception {
     Assert.assertTrue(outputValue.contains("CDAP,Platform"));
   }
 
+  @Test
+  public void testSkipHeader() throws Exception {
+    Schema schema = Schema.recordOf("user",
+                                    Schema.Field.of("body", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("offset", Schema.of(Schema.Type.LONG)),
+                                    Schema.Field.of("file", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    testSkipHeader("text", "", schema);
+
+    schema = Schema.recordOf("user",
+                             Schema.Field.of("val1", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("val2", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("val3", Schema.of(Schema.Type.INT)),
+                             Schema.Field.of("file", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    testSkipHeader("csv", ",", schema);
+    testSkipHeader("tsv", "\t", schema);
+    testSkipHeader("delimited", " ", schema);
+
+  }
+
+  private void testSkipHeader(String format, String delimeter, Schema schema) throws Exception {
+    File inputFile = temporaryFolder.newFile();
```

Comment:
better to write this once and pass it in as an argument.

Reply:
the content in the file is different for each delimeter, cannot be created once for all tests